main/src/ca/uwaterloo/flix/language/phase/extra/CodeHinter.scala

```diff
@@ -15,350 +16,170 @@
  */
 package ca.uwaterloo.flix.language.phase.extra
 
-import ca.uwaterloo.flix.api.Flix
-import ca.uwaterloo.flix.api.lsp.Index
-import ca.uwaterloo.flix.language.ast.TypedAst.Predicate.{Body, Head}
+import ca.uwaterloo.flix.api.lsp.acceptors.AllAcceptor
+import ca.uwaterloo.flix.api.lsp.{Consumer, Visitor}
 import ca.uwaterloo.flix.language.ast.TypedAst.*
-import ca.uwaterloo.flix.language.ast.shared.SymUse.DefSymUse
-import ca.uwaterloo.flix.language.ast.{Ast, SourceLocation, Symbol, Type, TypeConstructor, TypedAst}
+import ca.uwaterloo.flix.language.ast.shared.SymUse.{DefSymUse, TraitSymUse}
+import ca.uwaterloo.flix.language.ast.shared.{Annotations, SymUse, TraitConstraint}
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Symbol, Type, TypeConstructor, TypedAst}
 import ca.uwaterloo.flix.language.errors.CodeHint
 
 object CodeHinter {
 
   /**
     * Returns a collection of code quality hints for the given AST `root`.
     */
-  def run(root: TypedAst.Root, sources: Set[String])(implicit flix: Flix, index: Index): List[CodeHint] = {
-    val traitHints = root.traits.values.flatMap(visitTrait(_)(index)).toList
-    val defsHints = root.defs.values.flatMap(visitDef(_)(root, flix)).toList
-    val enumsHints = root.enums.values.flatMap(visitEnum(_)(index)).toList
-    (traitHints ++ defsHints ++ enumsHints).filter(include(_, sources))
+  def run(sources: Set[String])(implicit root: Root): List[CodeHint] = {
+    val cands = getCandidates
```

Comment:
why not something like: getOccurrences? or getUses?

Then the case class could be "Uses"

Reply:
I did originally call it Uses, but I figured it didn't really reflect what it was that cleanly, since it also contains def application stuff, which are not really "uses" as such.

Reply:
Occurrences then?